experimental/iterators/lib/Conversion/IteratorsToStandard/IteratorsToStandard.cpp

```diff
@@ -0,0 +1,159 @@
+//===- IteratorsToStandard.h - Conversion from Iterators to std dialect ---===//
+
+#include "iterators/Conversion/IteratorsToStandard/IteratorsToStandard.h"
+
+#include "../PassDetail.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMTypes.h"
+#include "mlir/Dialect/Utils/StructuredOpsUtils.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/TypeRange.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/IR/DerivedTypes.h"
+#include "llvm/Support/Casting.h"
+#include "llvm/Support/ScopedPrinter.h" // XXX
+
+#include <memory>
+
+using namespace mlir;
+using namespace mlir::iterators;
+
+namespace {
+struct ConvertIteratorsToStandardPass
+    : public ConvertIteratorsToStandardBase<ConvertIteratorsToStandardPass> {
+  void runOnOperation() override;
+};
+} // namespace
+
+/// Maps IteratorType to llvm.ptr<i8>
+class IteratorsTypeConverter : public TypeConverter {
+public:
+  IteratorsTypeConverter() {
+    addConversion([](Type type) { return type; });
+    addConversion(convertIteratorType);
+  }
+
+  static Optional<Type> convertIteratorType(Type type) {
+    if (type.isa<iterators::IteratorType>())
+      return LLVM::LLVMPointerType::get(IntegerType::get(type.getContext(), 8));
+    return llvm::None;
+  }
+};
+
+/// Returns or creates a function declaration at the module of the provided
+/// original op,
+FuncOp lookupOrCreateFuncOp(llvm::StringRef fnName, FunctionType fnType,
+                            Operation *op, PatternRewriter &rewriter) {
+  ModuleOp module = op->getParentOfType<ModuleOp>();
```

Comment:
Please assert ModuleOp is not empty.
In the future we may want a different op with a SymbolTable.